### PR TITLE
fix: fill in Rust error message templates

### LINKED ISSUE #1916: Fill in Rust error message templates
In several places, error message string templates aren't filled in: (regex: `\{:\?\}[^"]*"\)`)
```
rust/src/accuracy/mod.rs:
   36:         return fallible!(InvalidDistance, "alpha ({:?}) must be in (0, 1]");
   64:         return fallible!(InvalidDistance, "alpha ({:?}) must be in (0, 1]");
   98:         return fallible!(InvalidDistance, "alpha ({:?}) must be in (0, 1)");
  185:         return fallible!(InvalidDistance, "alpha ({:?}) must be in (0, 1]");
  241:         return fallible!(InvalidDistance, "alpha ({:?}) must be in (0, 1)");

rust/src/measurements/make_private_lazyframe/group_by/mod.rs:
  114:         return fallible!(MakeMeasurement, "The key-set of {:?} is private and cannot be released without filtering. Please pass a filtering threshold into make_private_lazyframe.");

rust/src/transformations/dataframe/apply/mod.rs:
  43:                 err!(FailedFunction, "{:?} does not exist in the input dataframe")

rust/src/transformations/dataframe/subset/mod.rs:
  36:                 .ok_or_else(|| err!(FailedFunction, "{:?} does not exist in the input dataframe"))?
  46:                     err!(FailedFunction, "{:?} does not exist in the input dataframe")
```

Taking one example:
```
return fallible!(MakeMeasurement, "The key-set of {:?} is private and cannot be released without filtering. Please pass a filtering threshold into make_private_lazyframe.");
```
Should be:
```
return fallible!(MakeMeasurement, "The key-set of {:?} is private and cannot be released without filtering. Please pass a filtering threshold into make_private_lazyframe.", grouping_columns);
```

To catch these at build time (from Mike):
```
// "error" is shadowed, and breaks intellij macro resolution
/// Create an instance of [`Error`]
#[macro_export]
macro_rules! err {
    // error without message
    ($variant:ident) => ($crate::error::Error {
        variant: $crate::error::ErrorVariant::$variant,
        message: None,
        backtrace: err!(@backtrace)
    });
    // error with explicit message
    ($variant:ident, $message:expr) => 
        (err!(@new $variant, format!($message)));
    // error with template formatting
    ($variant:ident, $template:expr, $($args:expr),+) =>
        (err!(@new $variant, format!($template, $($args,)+)));

    (@new $variant:ident, $message:expr) => ($crate::error::Error {
        variant: $crate::error::ErrorVariant::$variant,
        message: Some($message),
        backtrace: err!(@backtrace)
    });

    (@backtrace) => (std::backtrace::Backtrace::capture());
}
```

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes #1916

### CODE REVIEW

**Comment by Shoeboxam** [comment]
* **#1917** <a href="https://app.graphite.dev/github/pr/opendp/opendp/1917?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Shoeboxam and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by mccalluc** [comment]
Failing with:
```
error: format argument must be a string literal
   --> src/error/mod.rs:119:30
    |
119 |         err!(FailedFunction, v)
    |                              ^
    |
help: you might be missing a string literal to format with
    |
119 |         err!(FailedFunction, "{}", v)
    |                              +++++

error: could not compile `opendp` (lib test) due to 1 previous error
```

File: rust/src/error/mod.rs [hunk 1]
```diff
@@ -26,14 +26,17 @@ macro_rules! err {
         backtrace: err!(@backtrace)
     });
     // error with explicit message
-    ($variant:ident, $message:expr) => ($crate::error::Error {
+    ($variant:ident, $message:expr) =>
+        (err!(@new $variant, format!($message)));
```
Comment (mccalluc) [inline rust/src/error/mod.rs:30]: Out of my depth, but in this case, is it just a string? do we need `format!`

**Review by mccalluc — COMMENTED** [review]
If the CI passes, I'd be happy to approve.

**Review by mccalluc — APPROVED** [review]
Approved. 

- Down the road (#1830) I'd like to have more testing of exceptions in Python, and then we'd assert that the error messages look right.

**Pull request merged**